build.gradle.kts

```diff
@@ -62,164 +36,38 @@ dependencies {
-}
-
-tasks.withType<Copy> {
-    duplicatesStrategy = DuplicatesStrategy.INCLUDE
-}
-
-tasks.register<Test>("integrationTest") {
-    useJUnitPlatform()
-    description = "Runs the integration tests."
-    group = "verification"
-    testClassesDirs = sourceSets["integrationTest"].output.classesDirs
-    classpath = sourceSets["integrationTest"].runtimeClasspath
-    outputs.upToDateWhen { false }
-    mustRunAfter(tasks["test"])
-}
-
-// Configure Gradle Kover Plugin - read more: https://github.com/Kotlin/kotlinx-kover#configuration
-koverReport {
-    defaults {
-        xml {
-            onCheck = true
-        }
-    }
+    version.set("2023.3")
+    type.set("IC") // Target IDE Platform
+    plugins.set(listOf("com.redhat.devtools.intellij.telemetry:1.2.1.62", "textmate", "properties"))
 }
 
 tasks {
-    wrapper {
-        gradleVersion = properties("gradleVersion").get()
+    // Set the JVM compatibility versions
+    withType<JavaCompile> {
+        sourceCompatibility = "17"
+        targetCompatibility = "17"
     }
-
-    runPluginVerifier {
-        failureLevel = listOf(INVALID_PLUGIN, COMPATIBILITY_PROBLEMS, MISSING_DEPENDENCIES )
-        verificationReportsFormats = listOf(MARKDOWN, HTML)
+    withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
+        kotlinOptions.jvmTarget = "17"
     }
 
     patchPluginXml {
-        version = properties("pluginVersion")
-        sinceBuild = properties("pluginSinceBuild")
-        untilBuild = properties("pluginUntilBuild")
-        //TODO inject changelog into plugin.xml change-notes
-    }
-
-    // TODO: Once LSP4IJ no longer supports 2023.*, the hotswap JVM args can be added unconditionally
-    fun supportsEnhancedClassRedefinition(): Boolean {
-        val platformVersion = properties("platformVersion").getOrNull()
-        return when (platformVersion) {
-            // Versions before 2024.1 don't support enhanced class redefinition
-            "2023.2", "2023.3" -> false
-            else -> true
-        }
-    }
-
-    runIde {
-        // Improved hotswap for the IDE's JVM
-        if (supportsEnhancedClassRedefinition()) {
-            jvmArgs("-XX:+AllowEnhancedClassRedefinition", "-XX:HotswapAgent=fatjar")
-        }
-        //Use "debug" to send telemetry to dev source at segment.com
-        systemProperties["com.redhat.devtools.intellij.telemetry.mode"] = "debug" // "disabled"
-    }
-
-    // Configure UI tests plugin
-    // Read more: https://github.com/JetBrains/intellij-ui-test-robot
-    runIdeForUiTests {
-        systemProperty("robot-server.port", "8082")
-        systemProperty("ide.mac.message.dialogs.as.sheets", "false")
-        systemProperty("jb.privacy.policy.text", "<!--999.999-->")
-        systemProperty("jb.consents.confirmation.enabled", "false")
-    }
-
-    jacocoTestReport {
-        reports {
-            xml.required = true
-            html.required = true
-        }
+        sinceBuild.set("233")
+        untilBuild.set("243.*")
```

Comment:
Previously the `until-build` wasn't being set. This is setting it, though. I think we can safely remove this line and it will behave as it did previously, but I haven't done that yet here.